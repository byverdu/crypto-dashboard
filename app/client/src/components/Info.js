import React from 'react';
import { Alert } from 'reactstrap';

const Info = ({
  ...props, text, type, children
}) => {
  const className = props.fade ? 'hide' : '';

  return (
    <Alert color={type} className={className} >
      {text}
      {children && children}
    </Alert>
  );
};

export default Info;
