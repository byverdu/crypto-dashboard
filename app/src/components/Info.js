import React from 'react';
import { Alert } from 'reactstrap';

const Info = ({ text, type }) => <Alert color={type}>
  {text}
</Alert>;

export default Info;
