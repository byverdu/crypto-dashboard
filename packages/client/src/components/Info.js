import React from 'react';
import PropTypes from 'prop-types';
import { SnackbarContent } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { amber, blue } from '@material-ui/core/colors/';
import classNames from 'classnames';

const styles = theme => ({
  warning: {
    background: amber[ 700 ],
    color: 'black',
    fontSize: 20
  },
  error: {
    background: theme.palette.error.dark
  },
  info: {
    background: blue[ 600 ]
  }
});

const Info = ({
  message, type, className, classes, ...props
}) => (
  <SnackbarContent
    className={classNames( classes[ type ], className )}
    message={message || props.children}
  />
);

Info.propTypes = {
  message: PropTypes.string,
  type: PropTypes.oneOf(
    ['success', 'warning', 'error', 'info']
  ).isRequired
};

Info.defaultProps = {
  message: 'Kameha',
  type: 'success'
};

export default withStyles( styles )( Info );
