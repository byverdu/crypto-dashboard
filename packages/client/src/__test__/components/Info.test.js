import React from 'react';
import { shallow, mount } from 'enzyme';
import Info from '../../components/Info';
import renderer from 'react-test-renderer';

let wrapped;
let mounted;
const component = <Info type="error" message="xoxo" />;

beforeEach(() => {
  wrapped = shallow( component );
  mounted = mount(
    <Info type="info" message={null}><h1>Hello World</h1></Info>
  );
});

describe( 'Info component', () => {
  it( 'renders correctly', () => {
    const tree = renderer
      .create( component )
      .toJSON();
    expect( tree ).toMatchSnapshot();
  });

  it( 'should have a "type" prop', () => {
    expect( wrapped.props()).toHaveProperty( 'type' );
    expect( wrapped.props().type ).toEqual( 'error' );
  });

  it( 'should have a "message" prop', () => {
    expect( wrapped.props()).toHaveProperty( 'message' );
    expect( wrapped.props().message ).toEqual( 'xoxo' );
  });

  it( 'should override the "message" prop with children', () => {
    const children = mounted.find( 'h1' ).text();

    expect( children ).toEqual( 'Hello World' );
  });
});
