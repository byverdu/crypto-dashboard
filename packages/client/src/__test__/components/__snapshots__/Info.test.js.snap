// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component renders correctly 1`] = `
<div
  className="MuiTypography-root-33 MuiTypography-body1-42 MuiPaper-root-6 MuiPaper-elevation6-14 MuiSnackbarContent-root-3 Info-error-2"
  role="alertdialog"
>
  <div
    className="MuiSnackbarContent-message-4"
  >
    xoxo
  </div>
</div>
`;
